src/main/java/org/broadinstitute/barclay/help/HelpDoclet.java

```diff
@@ -89,6 +91,9 @@
      * @throws java.io.IOException if output can't be written.
      */
     protected boolean startProcessDocs(final RootDoc rootDoc) throws IOException {
+        // set the indexFileExtension to null to identify if it was provided with -index-file-extension option
+        // this is necessary the make the default the same as the -output-file-extension if it wasn't provided
+        indexFileExtension = null;
```

Comment:
Ok, I will keep them independent. I wanted to make it flexible and allow the user to only specify one of them for consistency unless specified. For example: everything in Markdown, unless you specify that the index should be an yml file.

Anyway, I removed this in the next commit.